subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/LoggingIntegrationTest.groovy

```diff
@@ -200,9 +202,22 @@ class LoggingIntegrationTest extends AbstractIntegrationTest {
 
     @Test
     public void multiThreadedDebugLogging() {
+        disableStacktraceCheckOnJava7()
         checkOutput(this.&runMultiThreaded, multiThreaded.debug)
     }
 
+    /**
+     * Sometimes on Java7, when another client tries to connect to the daemon, we
+     * get a spurious stacktrace in the debug output from the daemon related to the
+     * other connection.  It doesn't affect this test functionally, but the stacktrace
+     * check fails the test anyways because the logged error shows up in the debug output.
+     */
+    def disableStacktraceCheckOnJava7() {
```

Comment:
This is a work around, not a fix. What's the stacktrace look like? We should make that go away instead of having the test ignore the problem.

Reply:
See the issue attached to the PR.  I've already backed this out and implemented what I hope is a fix (https://github.com/gradle/gradle/commit/fef8f1f17120d0984b5f45c031a9fe06e02a8c27).  I've been unsuccessful at reproducing the problem, so I don't know if I've actually fixed it or not, but what I believe is happening is that the `DefaultIncomingConnectionHandler` thread that stops the `DefaultDaemonConnection` is stopping the underlying connection at just the right moment, which causes an exception during the read from the connection.  I've changed it so that we don't bother logging these read exceptions when we're in the process of tearing down the connection.